alerting/src/main/kotlin/org/opensearch/alerting/settings/AlertingSettings.kt

```diff
@@ -156,5 +170,70 @@ class AlertingSettings {
             -1L,
             Setting.Property.NodeScope, Setting.Property.Dynamic
         )
+
+        val METRICS_STORE_TIME = Setting.timeSetting(
+            METRICS_STORE_TIME_DEFAULT.key,
+            METRICS_STORE_TIME_DEFAULT,
+            StorageValidator(),
+            Setting.Property.NodeScope, Setting.Property.Dynamic
+        )
+
+        val METRICS_EXECUTION_FREQUENCY = Setting.timeSetting(
+            METRICS_EXECUTION_FREQUENCY_DEFAULT.key,
+            METRICS_EXECUTION_FREQUENCY_DEFAULT,
+            ExecutionValidator(),
+            Setting.Property.NodeScope, Setting.Property.Dynamic
+        )
+        internal class ExecutionValidator : Setting.Validator<TimeValue> {
+            override fun validate(value: TimeValue) {}
+
+            override fun validate(value: TimeValue, settings: Map<Setting<*>, Any>) {
+                val storageTime = settings[METRICS_STORE_TIME] as TimeValue
+                log.info("THIS IS STORAGETIME $storageTime")
+                validateSettings(value, storageTime)
+            }
+
+            override fun settings(): MutableIterator<Setting<*>> {
+                val settings = mutableListOf<Setting<*>>(
+                    METRICS_EXECUTION_FREQUENCY,
+                    METRICS_STORE_TIME
+                )
+                return settings.iterator()
+            }
+        }
+
+        internal class StorageValidator : Setting.Validator<TimeValue> {
+            override fun validate(value: TimeValue) {}
+
+            override fun validate(value: TimeValue, settings: Map<Setting<*>, Any>) {
+                val executionTime = settings[METRICS_EXECUTION_FREQUENCY] as TimeValue
+                log.info("THIS IS EXECUTIONTIME AS MILLIS ${executionTime.millis()}")
+                log.info("THIS IS EXECUTIONTIME $executionTime")
+                validateSettings(executionTime, value)
+            }
+
+            override fun settings(): MutableIterator<Setting<*>> {
+                val settings = mutableListOf<Setting<*>>(
+                    METRICS_EXECUTION_FREQUENCY,
+                    METRICS_STORE_TIME
+                )
+                return settings.iterator()
+            }
+        }
+        private fun validateSettings(executionFrequency: TimeValue, storageTime: TimeValue) {
+            if (executionFrequency > storageTime) {
+                throw IllegalArgumentException(
+                    "The execution frequency should be less than the storage time."
+                )
+            } else if (executionFrequency < MINIMUM_EXECUTION_FREQUENCY) {
+                throw IllegalArgumentException(
+                    "The execution frequency can not be less than 1 second."
```

Comment:
Call your constant here instead of typing out `1 second`. That will help keep the exception's message aligned with the minimum value constant.